### DISCUSSION TITLE
Automate Tests with GitHub Actions

### DISCUSSION THREAD

**Discussion by vicainelli** [adonisjs/core#1930] [Help] [answered]
Hey all, I'm writing a scaffold API project with AdonisJS, and I'm facing a issues on set up the GitHub actions pipeline.

First, I got and exception. but it was "solved" after I hard coded some key on APP_KEY at `.env.example` 
```
Exception: E_MISSING_ENV_VALUE: Missing environment variable "APP_KEY"
```

And after thatI got a connection error
```
[ error ]  SQLITE_CANTOPEN: unable to open database file
```

Here's the `.yml` configuration file: https://github.com/vicainelli/node-scaffold-api-adonisjs/blob/principal/.github/workflows/test.yml

And here's the failed pipeline: https://github.com/vicainelli/node-scaffold-api-adonisjs/runs/1419752738?check_suite_focus=true

**Comment by vicainelli** [comment, marked as answer]
Ok, for some reason I still don't know, the database are stored in the `tmp` directory.

I just added a command to create this directory and it works.

But I'm still facing the "problem" with the APP_KEY.

**Reply by thetutlage** [reply]
You need to define the environment variables. I am not sure there is any way to get around it.

**Reply by vicainelli** [reply]
Yes, I noticed, even in the local environment, I have to copy and paste the key-value after run `node ace generate:key`, it's not filled automatically.

**Reply by thetutlage** [reply]
Yes, because keeping environment variables inside the `.env` file is one of the many ways. In CI, I will define them inside the CI web interface, in production I may pull it from somewhere else
